cime/src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -3651,6 +3676,47 @@ subroutine cime_run_atmocn_setup(hashint)
 
   end subroutine cime_run_atmocn_setup
 
+!----------------------------------------------------------------------------------
+
+  subroutine cime_run_ocnglc_coupling()
+    !---------------------------------------
+    ! Description: Run calculation of coupling fluxes between OCN and GLC
+    !    Note: this happens in the coupler to allow it be calculated on the
+    !    ocean time step but the GLC grid.
+    !---------------------------------------
+
+    if (glc_present) then
+
+          if (ocn_c2_glc) then
+             call prep_glc_calc_o2x_gx(timer='CPL:glcprep_ocn2glc') !remap ocean fields to o2x_g at atmospheric timestep
+          end if
+
+          if (glc_c2_ocn) then
+             call prep_glc_calculate_subshelf_boundary_fluxes! this outputs
+                                           !x2g_g/g2x_g, where latter is going
+                                           !to ocean, so should get remapped to
+                                           !ocean grid in prep_ocn_shelf_calc_g2x_ox
+             call prep_ocn_shelf_calc_g2x_ox(timer='CPL:glcpost_glcshelf2ocn')
+                                           !Map g2x_gx shelf fields that were updated above, to g2x_ox.
+                                           !Do this at intrinsic coupling
+                                           !frequency
+             call prep_ice_shelf_calc_g2x_ix(timer='CPL:glcpost_glcshelf2ice')
+                                           !Map g2x_gx shelf fields to g2x_ix.
+                                           !Do this at intrinsic coupling
+                                           !frequency.  This is perhaps an
+                                           !unnecessary place to put this
+                                           !call, since these fields aren't
+                                           !changing on the intrinsic
+                                           !timestep.  But I don't think it's
+                                           !unsafe to do it here.
+          endif
+
+          call prep_glc_accum_ocn(timer='CPL:glcprep_accum_ocn') !accum x2g_g fields here into x2g_gacc, along with l2gacc_lx
```

Comment:
Should this call be in a conditional beyond just `glc_present`? Maybe `ocn_c2_glc`?

Reply:
@billsacks , yes, good point.  I've added that in a new commit (312aa4fb348f8b5584ae612a4692a81860d5eb65).  I actually changed the conditional logic for the entire routine on further review, and I'm hoping to get blessing from @JeremyFyke on that change.